pallets/phala/src/migrations/mod.rs

```diff
@@ -120,3 +140,263 @@ pub mod v6 {
+	{
+		stakepool::pallet::StakeLedger::<T>::drain().for_each(|(user_id, balance)| {
+			<T as PhalaConfig>::Currency::remove_lock(STAKING_ID, &user_id);
+			<T as PhalaConfig>::Currency::remove_lock(VESTING_ID, &user_id);
+			<T as PhalaConfig>::Currency::remove_lock(DEMOCRAC_ID, &user_id);
+			<T as PhalaConfig>::Currency::remove_lock(PHRELECT_ID, &user_id);
+			if <T as PhalaConfig>::Currency::transfer(
+				&user_id,
+				&<T as pawnshop::Config>::PawnShopAccountId::get(),
+				balance,
+				KeepAlive,
+			)
+			.is_err()
+			{
+				<T as PhalaConfig>::Currency::transfer(
+					&computation::Pallet::<T>::account_id(),
+					&user_id,
+					<T as PhalaConfig>::Currency::minimum_balance(),
+					KeepAlive,
+				);
+				<T as PhalaConfig>::Currency::transfer(
+					&user_id,
+					&<T as pawnshop::Config>::PawnShopAccountId::get(),
+					balance,
+					KeepAlive,
+				)
+				.expect("transfer should not fail; qed.");
+			}
+		});
+	}
+
+	fn migrate_unchanged_stuffs<T: PhalaPallets>()
+	where
+		T: PhalaPallets + stakepool::pallet::Config + PhalaConfig,
+		BalanceOf<T>: balance_convert::FixedPointConvert + sp_std::fmt::Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32>,
+		T: pallet_assets::Config<Balance = BalanceOf<T>>,
+	{
+		stakepool::pallet::WorkerAssignments::<T>::drain().for_each(|(k, v)| {
+			stakepoolv2::pallet::WorkerAssignments::<T>::insert(k, v);
+		});
+		stakepool::pallet::SubAccountPreimages::<T>::drain().for_each(|(k, v)| {
+			stakepoolv2::pallet::SubAccountPreimages::<T>::insert(k, v);
+		});
+		stakepool::pallet::PoolContributionWhitelists::<T>::drain().for_each(|(k, v)| {
+			stakepoolv2::pallet::PoolContributionWhitelists::<T>::insert(k, v);
+		});
+		stakepool::pallet::PoolDescriptions::<T>::drain().for_each(|(k, v)| {
+			stakepoolv2::pallet::PoolDescriptions::<T>::insert(k, v);
+		});
+	}
+
+	#[cfg(feature = "try-runtime")]
+	pub fn pre_migrate<T: PhalaPallets>() -> Result<(), &'static str> {
+		frame_support::ensure!(
+			get_versions::<T>() == unified_versions::<T>(6),
+			"incorrect pallet versions"
+		);
+		Ok(())
+	}
+
+	pub fn migrate<T>() -> Weight
+	where
+		T: PhalaPallets + stakepool::pallet::Config + PhalaConfig,
+		BalanceOf<T>: balance_convert::FixedPointConvert + sp_std::fmt::Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32>,
+		T: pallet_assets::Config<Balance = BalanceOf<T>>,
+	{
+		if get_versions::<T>() == unified_versions::<T>(6) {
+			let mut weight: Weight = Weight::zero();
+			log::info!("Ᵽ migrating phala-pallets to v7");
+			weight += stakepoolv2::Pallet::<T>::migration_remove_assignments();
+			log::info!("Ᵽ pallets migrated to v7");
+
+			migrate_stake_pools::<T>();
+			migrate_pool_stakers::<T>();
+			migrate_stake_ledger::<T>();
+			migrate_unchanged_stuffs::<T>();
+			set_unified_version::<T>(7);
```

Comment:
Let's try our best to account the weights in each migration steps. Weights are used to measure the cost of reading & writing

Reply:
We will do it after migration tests done.